src/App.tsx

```diff
@@ -17,18 +37,33 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                setVisibleTodoList={setVisibleTodoList}
+                todoListFromServer={todos}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {todos.length > 0
+                ? (
+                  <TodoList
+                    todoList={visibleTodoList}
+                    selectedTodo={selectedTodo}
+                    setSelectedTodo={setSelectedTodo}
+                  />
+                )
+                : (<Loader />)}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {selectedTodo.id && (
+        <TodoModal
+          selectedTodo={selectedTodo}
+          setSelectedTodo={setSelectedTodo}
+        />
+      )}
```

Comment:
![image](https://user-images.githubusercontent.com/102293862/200363246-a13f1ece-2372-4ad9-b47a-6d9893723055.png)
There's "0" at the bottom of the page. It's because if selectedTodo.id is string zero than it evaluates to true and just render 0.